### PR TITLE
LLP-65-BurgerMenu

### PR DESCRIPTION
**Pull request by mdainys**

### CODE REVIEW

File: src/components/BurgerMenu.js [hunk 1]
```diff
@@ -1,66 +1,256 @@
...
-        <span></span>
-        <span></span>
-        <span></span>
-      </label>
-      <ul className="menu">
-        <li>
-          <Link className={styles.burgerLink} href="/payments">
-            Payment
+    <div
+      ref={ref}
```
Comment (pinenetcoder) [inline src/components/BurgerMenu.js:41]: delete if not neccessary

File: src/components/BurgerMenu.js [hunk 2]
```diff
@@ -1,66 +1,256 @@
...
+      >
+        {langBtnState}
+      </span>
     </div>
   );
-}
+};
 
-export default BurgerMenu;
+export default React.forwardRef(BurgerMenu);
```
Comment (pinenetcoder) [inline src/components/BurgerMenu.js:256]: Why we need React.forwardRef here ?

File: src/components/Drawer.js [hunk 3]
```diff
@@ -0,0 +1,28 @@
+import React, { useRef } from "react";
```
Comment (pinenetcoder) [inline src/components/Drawer.js:1]: No nedd for ref. Pleae Delete all related code

File: src/components/Drawer.js [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+    <div>
+      <Image
+        src="/assets/images/burgerIcon.svg"
+        width={20}
+        height={20}
+        alt="Menu"
+        onClick={handleToggleBurgerMenu}
+        className="burgerIcon"
+      />
+      <BurgerMenu ref={burgerMenuRef} isOpen={isOpen} onClose={onClose} />
```
Comment (pinenetcoder) [inline src/components/Drawer.js:23]: No need for ref

File: src/components/Header.js [hunk 5]
```diff
@@ -1,141 +1,207 @@
 import Image from "next/image";
 import Link from "next/link";
 import { useRouter } from "next/router";
-import { useEffect, useState } from "react";
-import lt from '@/locales/lt';
-import en from '@/locales/en';
-import BurgerMenu from "./BurgerMenu";
+import { useEffect, useState, useRef } from "react";
```
Comment (pinenetcoder) [inline src/components/Header.js:4]: Please delete useRef

File: src/components/Header.js [hunk 6]
```diff
@@ -1,141 +1,207 @@
...
+import lt from "@/locales/lt";
+import en from "@/locales/en";
+import Drawer from "./Drawer";
 
 function Header() {
-  const [langBtnState, setLangBtnState] = useState('ENG');
+  const [isMenuOpen, setIsMenuOpen] = useState(false);
+  const drawerRef = useRef(null);
```
Comment (pinenetcoder) [inline src/components/Header.js:11]: Please delete unused Code

Comment (mdainys) [reply src/components/Header.js:11]: done

File: src/components/Header.js (hunk 5 above)
Comment (mdainys) [reply src/components/Header.js:4]: done

File: src/pages/terms-and-conditions.js [hunk 7]
```diff
@@ -6,21 +6,53 @@ import IndexLayout from "@/Layouts/IndexLayout";
...
-      let tncList = [];
-      snapshot.docs.forEach((tab) => {
-        tncList.push({ ...tab.data(), id: tab.id });
-      });
-      console.log(tncList);
-    })
-    .catch((err) => {
-      console.log(err.message);
-    });
+  // const db = getFirestore();
```
[the remark this answers is not in the document]
Comment (mdainys) [reply src/pages/terms-and-conditions.js:12]: got it !

File: src/components/Drawer.js (hunk 4 above)
Comment (mdainys) [reply src/components/Drawer.js:23]: done

File: src/components/Drawer.js [hunk 8]
```diff
@@ -0,0 +1,28 @@
+import React, { useRef } from "react";
+import styles from "@/styles/BurgerMenu.module.scss";
+import Image from "next/image";
+import BurgerMenu from "./BurgerMenu";
+
+const Drawer = ({ isOpen, onClose }) => {
+  const burgerMenuRef = useRef(null);
```
[the remark this answers is not in the document]
Comment (mdainys) [reply src/components/Drawer.js:7]: done

File: src/components/Drawer.js (hunk 3 above)
Comment (mdainys) [reply src/components/Drawer.js:1]: done

File: src/components/BurgerMenu.js (hunk 2 above)
Comment (mdainys) [reply src/components/BurgerMenu.js:256]: done

File: src/components/BurgerMenu.js (hunk 1 above)
Comment (mdainys) [reply src/components/BurgerMenu.js:41]: done

**Review by pinenetcoder — APPROVED** [review]

**Pull request merged**